cmd/release-controller/upgrades.go

```diff
@@ -345,3 +349,96 @@ func syncGraphToSecret(graph *UpgradeGraph, update bool, secretClient kv1core.Se
 		glog.V(2).Infof("Saved upgrade graph state to %s/%s", ns, name)
 	}, 5*time.Minute, stopCh)
 }
+
+func releaseTagTimestamp(v string) (time.Time, error) {
+	var t time.Time
+	version, err := semverParseTolerant(v)
+	if err != nil {
+		return t, err
+	}
+	if len(version.Pre) == 0 {
+		return t, fmt.Errorf("tag %s has no timestamp", v)
+	}
+	prereleaseLast := version.Pre[len(version.Pre) - 1].VersionStr
+	releaseTagTimestampFormat := "2006-01-02-150405"
+	if len(prereleaseLast) < len(releaseTagTimestampFormat) {
```

Comment:
this is questionable - you should never have to check against anything other than "len(prereleaseLast) == 0" because you shouldn't be guessing at the value.

Reply:
This was mostly because the timestamp format needs that length to parse.